NewForce-Cohort-9/tabloid-full-stack-code-crusaders
# Df branch - #13 View All Tags

https://trello.com/c/fqZasKcm - #13 View All Tags

**Description**
This PR implements an admin feature that allows users to view tags in the Tag Management section of the TabloidFullStack application. All tags are displayed in alphabetical order to improve navigation and management for admins. For the time being, all users are treated as admins, with future stories planned to enforce user permissions.

**Type of Change**
Enhancement (non-breaking change that improves existing functionality)

**Changes Made**
1. Tag Management Page:
- Created a page that lists all tags in alphabetical order.
2. Controller Methods:
- Added a Get method in the TagController to retrieve all tags in alphabetical order.
3. Frontend Changes:
- Created a new view for Tag Management.
- Updated the navigation menu to include a "Tag Management" link.

**Testing Instructions**
1. Preparation:
- Ensure all your local changes are committed and pushed to your branch.
- Switch to the main branch and update it with the latest changes:
- Ensure the latest version of the main branch is pulled (git pull origin main).
- git fetch --a will give you access to my branch "df-branch"
- git checkout df-branch to switch to my branch
_If you don't see the most recent version of my files, try git pull origin df-branch_
- If you haven't already created the Tabloid database, please do so using the provided SQL scripts.
- Use this [https://github.com/NewForce-Cohort-9/tabloid-full-stack-code-crusaders/blob/main/SQL/01_Tabloid_Create_DB.sql](url) to define the necessary tables and columns.
- Use this [https://github.com/NewForce-Cohort-9/tabloid-full-stack-code-crusaders/blob/main/SQL/02_Tabloid_Seed_Data.sql](url) to populate the database with initial data.

2. Launch the Application:
- After ensuring your environment is properly set up, launch the application by clicking the "Start with Debugger" (green play button) in your IDE.

3. Log In:
- Open the application in your browser.
- Navigate to the Login page via the navbar and log in with the following credentials:
Email: "foo@bar.com"
Password: Use any password of your choosing.
- After logging in, you should be redirected to the homepage.

4. Navigate to Tag Management:
- From the homepage, click on the Tag Management link in the navbar.
- You should be presented with a list of tags sorted in alphabetical order.

5. Verify:
- Confirm that the list displays the tag names in alphabetical order.
_Note that the Edit and Delete buttons are currently placeholders and do not have functionality at this time._

**Checklist**:
